File: zettle_ios/ios/Classes/ZettlePluginResponse.swift


/// Represents the general plugin response structure.
public struct ZettlePluginResponse {
    var methodName: String
    var success: Bool
    var response: [String: Any?]?
    var errorMessage: String?
    
    
    /// Converts the `ZettlePluginResponse` instance to a dictionary.
    /// - Returns: A dictionary representation of the plugin response.
    func toDict() -> [String: Any?]{
        return [
            "methodName": methodName,
            "success": success,
            "response": response,
            "errorMessage": errorMessage
        ]
    }
}

File: zettle_platform_interface/lib/models/zettle_plugin_response.dart
class ZettlePluginResponse {
  ZettlePluginResponse.fromMap(Map<dynamic, dynamic> response)
      : methodName = response['methodName'],
        success = response['success'],
        response = response['response'],
        errorMessage = response['errorMessage'];

  String methodName;
  String? errorMessage;
  bool success;
  Map<dynamic, dynamic>? response;

  @override
  String toString() {
    return 'Method: $methodName, success: $success, response: $response, errorMessage: $errorMessage';
  }
}
